action.yml

```diff
@@ -109,31 +129,30 @@ runs:
       shell: bash
       working-directory: ${{ inputs.working-directory }}
       run: |
-        resolver_nightly=
-
-        # Extra arguments for build and test steps, not the dependencies step
-        stack_build_arguments=()
-
-        if ${{ inputs.fast }}; then
-          stack_build_arguments+=( --fast )
-        fi
-
-        if ${{ inputs.pedantic }}; then
-          stack_build_arguments+=( --pedantic )
-        fi
+        {
+          echo "stack-query-arguments=              $(echo "${{ inputs.query-arguments       }}                                                 " | tr '\n' ' ')"
+          echo "stack-path-arguments=               $(echo "${{ inputs.path-arguments        }}                                                 " | tr '\n' ' ')"
+          echo "stack-setup-arguments=              $(echo "${{ inputs.setup-arguments       }}                                                 " | tr '\n' ' ')"
+          echo "stack-build-arguments-dependencies= $(echo "${{ inputs.stack-build-arguments }} ${{ inputs.stack-build-arguments-dependencies }}" | tr '\n' ' ')"
+          echo "stack-build-arguments-build=        $(echo "${{ inputs.stack-build-arguments }} ${{ inputs.stack-build-arguments-build        }}" | tr '\n' ' ')"
+          echo "stack-build-arguments-test=         $(echo "${{ inputs.stack-build-arguments }} ${{ inputs.stack-build-arguments-test         }}" | tr '\n' ' ')"
+        } >> "$GITHUB_OUTPUT"
 
         has_resolver() {
           grep -Fq -- '--resolver' <<'EOM'
         ${{ inputs.stack-arguments }}
         EOM
         }
 
+        resolver_nightly=
+
         if ! has_resolver && [[ "${{ inputs.stack-yaml }}" == 'stack-nightly.yaml' ]]; then
           resolver_nightly='--resolver nightly'
         fi
 
         echo "resolver-nightly=$resolver_nightly" >>"$GITHUB_OUTPUT"
-        echo "stack-build-arguments=${stack_build_arguments[*]}" >>"$GITHUB_OUTPUT"
+
+        echo "stack-arguments=--no-terminal --stack-yaml ${{ inputs.stack-yaml }} $resolver_nightly $(echo "${{ inputs.stack-arguments }}" | tr '\n' ' ')" >>"$GITHUB_OUTPUT"
```

Comment:
I'm not 100% sure that `--no-terminal` is actually necessary any more. I think GitHub has since done something to the runner machinery that you can emit terminal escapes and have it work correctly -- and actually be preferred because I think `--no-terminal` will strip colors? I'm not sure of that either.

Anyway, assuming we want to keep `--no-terminal`, let's make it the default for `stack-arguments`.

This way it's ensured to be consistent for every `stack` invocation, and it can be changed if desired.

Reply:
done in https://github.com/freckle/stack-action/pull/35/commits/513603eece22eb0c73619bbac7e33cb535f6d3d1